velox/functions/sparksql/tests/ArrayInsertTest.cpp

```diff
@@ -0,0 +1,122 @@
+    options_.parseIntegerAsBigint = false;
+  }
+
+  void testExpression(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& expected) {
+    const auto result = evaluate(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(ArrayInsertTest, nullSrcArrays) {
+  const auto arrays = makeArrayVectorFromJson<int64_t>({"null"});
+
+  const auto expected = makeArrayVectorFromJson<int64_t>({"null"});
+
+  testExpression("array_insert(c0, 1, 1, false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, nullPosition) {
+  const auto arrays = makeArrayVectorFromJson<int64_t>({"[1, 1]"});
+
+  const auto expected = makeArrayVectorFromJson<int64_t>({"null"});
+
+  testExpression(
+      "array_insert(c0, cast(null as integer), 1, false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, basic) {
+  const auto arrays = makeArrayVectorFromJson<int64_t>({"[1]", "[2, 2]"});
+
+  const auto expected =
+      makeArrayVectorFromJson<int64_t>({"[0, 1]", "[0, 2, 2]"});
+  testExpression("array_insert(c0, 1, 0, false)", {arrays}, expected);
+
+  const auto expected1 =
+      makeArrayVectorFromJson<int64_t>({"[null, 1]", "[null, 2, 2]"});
+  testExpression(
+      "array_insert(c0, 1, cast(null as integer), false)", {arrays}, expected1);
+}
+
+TEST_F(ArrayInsertTest, posGTArraySize) {
+  const auto arrays = makeArrayVectorFromJson<int64_t>({"[1]", "[2, 2]"});
+
+  const auto expected =
+      makeArrayVectorFromJson<int64_t>({"[1, null, 0]", "[2, 2, 0]"});
+  testExpression("array_insert(c0, 3, 0, false)", {arrays}, expected);
+
+  const auto expected1 =
+      makeArrayVectorFromJson<int64_t>({"[1, null, null]", "[2, 2, null]"});
+  testExpression(
+      "array_insert(c0, 3, cast(null as integer), false)", {arrays}, expected1);
+}
+
+TEST_F(ArrayInsertTest, negativePos) {
+  const auto arrays =
+      makeArrayVectorFromJson<int64_t>({"[1]", "[2, 2]", "[3, 3, 3]"});
+
+  const auto expected = makeArrayVectorFromJson<int64_t>(
+      {"[0, null, 1]", "[0, 2, 2]", "[3, 0, 3, 3]"});
+  testExpression("array_insert(c0, -3, 0, false)", {arrays}, expected);
+
+  const auto expected1 =
+      makeArrayVectorFromJson<int64_t>({"[1, 0]", "[2, 2, 0]", "[3, 3, 3, 0]"});
+  testExpression("array_insert(c0, -1, 0, false)", {arrays}, expected1);
+
+  const auto expected2 = makeArrayVectorFromJson<int64_t>(
+      {"[null, null, 1]", "[null, 2, 2]", "[3, null, 3, 3]"});
+  testExpression(
+      "array_insert(c0, -3, cast(null as integer), false)",
+      {arrays},
+      expected2);
+}
+
+TEST_F(ArrayInsertTest, negativePosLegacy) {
+  const auto arrays =
+      makeArrayVectorFromJson<int64_t>({"[1]", "[2, 2]", "[3, 3, 3]"});
+
+  const auto expected = makeArrayVectorFromJson<int64_t>(
+      {"[0, null, null, 1]", "[0, null, 2, 2]", "[0, 3, 3, 3]"});
+  testExpression("array_insert(c0, -3, 0, true)", {arrays}, expected);
+
+  const auto expected1 =
```

Comment:
Nit: 
We can just make `expected` re-usable. No need to define expected1, expected2...

Reply:
Sure, done.